service/controller/resource/promxy/resource.go

```diff
@@ -0,0 +1,152 @@
+	}
+
+	r := &Resource{
+		logger:       config.Logger,
+		k8sClient:    config.K8sClient,
+		installation: config.Installation,
+		provider:     config.Provider,
+	}
+
+	return r, nil
+}
+
+func (r *Resource) Name() string {
+	return "promxy"
+}
+
+func (r *Resource) toServerGroup(cluster metav1.Object) (*ServerGroup, error) {
+	httpClient := config.HTTPClientConfig{
+		TLSConfig: config.TLSConfig{
+			CAFile:             "/var/run/secrets/kubernetes.io/serviceaccount/ca.crt",
+			InsecureSkipVerify: true,
+		},
+		BearerTokenFile: "/var/run/secrets/kubernetes.io/serviceaccount/token",
+	}
+
+	apiServerHost := r.k8sClient.RESTConfig().Host
+	apiServerURL, err := url.Parse(apiServerHost)
+	if err != nil {
+		return nil, microerror.Mask(err)
+	}
+
+	return &ServerGroup{
+		AntiAffinity:   time.Second * 10,
+		Scheme:         "http",
+		RemoteReadPath: "api/v1/read",
+		RemoteRead:     true,
+		HTTPConfig: HTTPClientConfig{
+			DialTimeout: time.Millisecond * 200, // Default dial timeout of 200ms
+		},
+		Labels: model.LabelSet{
+			model.LabelName("installation"):     model.LabelValue(r.installation),
+			model.LabelName(key.ClusterIDKey()): model.LabelValue(key.ClusterID(cluster)),
+			model.LabelName("provider"):         model.LabelValue(r.provider),
+		},
+		PathPrefix: fmt.Sprintf("/%s", key.ClusterID(cluster)),
+		RelabelConfigs: []*relabel.Config{
+			{
+				SourceLabels: []model.LabelName{
+					"__meta_kubernetes_pod_label_app_kubernetes_io_managed_by",
+					"__meta_kubernetes_pod_label_app_kubernetes_io_name",
+					"__meta_kubernetes_pod_label_app_kubernetes_io_instance",
+				},
+				Action:    "keep",
+				Separator: ";",
+				Regex:     relabel.MustNewRegexp(fmt.Sprintf("prometheus-meta-operator;prometheus;%s", key.ClusterID(cluster))),
+			},
+		},
+		KubernetesSDConfigs: []*kubernetes.SDConfig{
+			&kubernetes.SDConfig{
+				APIServer: config.URL{
+					URL: apiServerURL,
+				},
+				Role:             kubernetes.RolePod,
+				HTTPClientConfig: httpClient,
+				NamespaceDiscovery: kubernetes.NamespaceDiscovery{
+					Names: []string{
+						key.Namespace(cluster),
+					},
+				},
+			},
+		},
+	}, nil
+}
+
+func (r *Resource) getConfigMap(ctx context.Context, obj interface{}) (*v1.ConfigMap, error) {
+	r.logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("checking if %s config map already exists ", key.PromxyConfigMapName()))
+	configMap, err := r.k8sClient.K8sClient().CoreV1().ConfigMaps(key.PromxyConfigMapNamespace()).Get(ctx, key.PromxyConfigMapName(), metav1.GetOptions{})
+
+	if apierrors.IsNotFound(err) {
+		r.logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("config map %s does not exists", key.PromxyConfigMapName()))
+		return nil, nil
+	} else if err != nil {
+		return nil, microerror.Mask(err)
+	}
+
+	return configMap, nil
+}
+
+func (r *Resource) readFromConfig(configMap *v1.ConfigMap) (*Promxy, error) {
+	content := configMap.Data[key.PromxyConfigFileName()]
```

Comment:
It would be safer to check if the key exists before accessing it to avoid panic.